File: backend/src/main/java/com/g0urd/grad_project/service/TransportationOfferService/TransportationOfferService.java
package com.g0urd.grad_project.service.TransportationOfferService;

import java.util.List;

import com.g0urd.grad_project.models.TransportationOffer;

public interface TransportationOfferService {

    List<TransportationOffer> fetchAllTransportationOffers();

    TransportationOffer findById(Long id);

    TransportationOffer createTransportationOffer(TransportationOffer transportationOffer);

    TransportationOffer updateTransportationOffer(TransportationOffer transportationOffer);

    Boolean deleteTransportationOffer(Long id);

}

File: frontend/src/app/service/transportation-offer/transportation-offer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TransportationOffer } from '../../component/transportation-offer/transportation-offer';

const BASE_URL = ["http://localhost:8080/api/manager"]

@Injectable({
  providedIn: 'root'
})
export class TransportationOfferService {

  constructor(private http: HttpClient) { }

  fetchAllTransportationOffers():Observable<TransportationOffer[]> {
    return this.http.get<TransportationOffer[]>(BASE_URL + `/transportation-offer`);
  }

  createTransportationOffer(data: TransportationOffer) {
    return this.http.post<TransportationOffer>(BASE_URL + `/transportation-offer`, data);
  }

  updateTransportationOffer(data: TransportationOffer) {
    return this.http.put<TransportationOffer>(BASE_URL + `/transportation-offer/${data.id_offer}`, data);
  }

  deleteTransportationOffer(id_offer: Number) {
    return this.http.delete<TransportationOffer>(BASE_URL + `/transportation-offer/${id_offer}`);
  }

}
